exonum-java-binding/testkit/src/main/java/com/exonum/binding/testkit/TestKit.java

```diff
@@ -206,10 +205,43 @@ public Block createBlockWithTransactions(Iterable<TransactionMessage> transactio
    * @return created block
    */
   public Block createBlock() {
+    List<TransactionMessage> inPoolTransactions = findTransactionsInPool(transactionMessage -> true);
+    checkTransactions(inPoolTransactions);
     byte[] block = nativeCreateBlock(nativeHandle.get());
     return BLOCK_SERIALIZER.fromBytes(block);
   }
 
+  private void checkTransactions(List<TransactionMessage> transactionMessages) {
+    for (TransactionMessage transactionMessage: transactionMessages) {
+      checkTransaction(transactionMessage);
+    }
+  }
+
+  private void checkTransaction(TransactionMessage transactionMessage) {
+    short serviceId = transactionMessage.getServiceId();
+    if (!services.containsKey(serviceId)) {
+      String message = String.format("Unknown service id (%s) in transaction message (%s)",
+          serviceId, transactionMessage);
+      throw new IllegalArgumentException(message);
+    }
+    Service service = services.get(serviceId);
+    try {
+      RawTransaction rawTransaction = toRawTransaction(transactionMessage);
+      service.convertToTransaction(rawTransaction);
+    } catch (Throwable conversionError) {
+      String message = String.format("Transaction message (%s) is invalid", transactionMessage);
```

Comment:
I specifically requested this error message to be _actionable_, and described the possible cases. Please fix the error message to reflect them.